### ISSUE TITLE
Problems under KDE Plasma

### ISSUE THREAD

**Issue by mard** [maximalmax90/vs1clickmodinstaller#1]
I have problems getting this to work correctly. Can you please try invoking the following command on your machine?

```
xdg-open "vintagestorymodinstall://primitivesurvival@3.5.5"
```

On my setup, xdg-open is turning "3.5.5" into "3.5.0.5" (4-octet IP host), rendering it impossible to download. Opening ModDB links with Chromium gives exactly the same result:

![Screenshot_20240517_142230](https://github.com/maximalmax90/vs1clickmodinstaller/assets/864964/4cc119f8-c390-4121-a4c0-940380522848)

![Screenshot_20240517_142733](https://github.com/maximalmax90/vs1clickmodinstaller/assets/864964/4af6e60b-ad9e-47f4-928c-e67c60d567bf)

It happens to work only with Firefox, maybe is uses a different implementation for URI parsing.

I had a full write-up about it last year in this thread:

https://www.vintagestory.at/forums/topic/10792-add-an-rfc-3986-compliant-alternative-to-vintagestorymodinstall-uri-for-linux-compatibility

If you can also reproduce this behavior with both xdg-open and Chromium based browsers, I will report an issue to Anego Studios and ask them to change the handler that will be compliant with URI specification and make it compatible with Freedesktop implementation.

**Comment by maximalmax90** [comment]
Checked via command line, firefox-based browser and chromium browser - all correct, video attached. I can't even tell what the problem might be.... Maybe some extension in the browser or software in the system is interfering.

https://github.com/maximalmax90/vs1clickmodinstaller/assets/75565218/cc162fe4-d395-4a22-87bd-2bf639d39533

**Comment by mard** [comment]
Thank you for help. I noticed that you're using a different desktop environment and it gave me an idea. Turns out it's a problem specific at least to KDE Plasma - if it detects KDE then `xdg-open` passes the argument to `kde-open` which causes the problem, while a generic implementation does not.

![Screenshot_20240518_130518](https://github.com/maximalmax90/vs1clickmodinstaller/assets/864964/035af38a-13b0-4a59-8851-15422da92a17)

I don't think it's a bug on KDE side either because they comply to the URI spec and they essentially have a full right to parse the number in authority component as a host.

Now I'm in a bit of a pickle. I've been looking to [ship the desktop entry for vintagestorymodinstall handler the with Vintage Story Flatpak package](https://github.com/flathub/at.vintagestory.VintageStory/issues/40) for quite some time, but obviously it won't work for Plasma users out-of-box. And I don't think Anego will bother fixing their URI because of a single specific Linux DE impacting a promile of users. I will need to think of a workaround for this.

Thanks again, I think that concludes this issue.
